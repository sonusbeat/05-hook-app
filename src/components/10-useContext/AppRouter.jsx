import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import HomeScreen from './HomeScreen';
import AboutScreen from "./AboutScreen";
import LoginScreen from './LoginScreen';
import NavBar from "./NavBar";

const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />

        <Switch>
          <Route exact path="/" component={ HomeScreen } />
          <Route exact path="/about" component={ AboutScreen } />
          <Route exact path="/login" component={ LoginScreen } />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  );
}

export default AppRouter;