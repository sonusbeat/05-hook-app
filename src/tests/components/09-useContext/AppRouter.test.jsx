import { mount } from 'enzyme';
import AppRouter from '../../../components/10-useContext/AppRouter';
import UserContext from '../../../components/10-useContext/UserContext';

describe('Pruebas en <AppRouter />', () => {
  const user = {
    id: 1,
    name: "Daniel",
    email: "user@domain.com"
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <AppRouter />
    </UserContext.Provider>
  );

  test('Deberia coincidir con el Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
