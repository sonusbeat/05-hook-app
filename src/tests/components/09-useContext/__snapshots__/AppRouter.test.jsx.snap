// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Deberia coincidir con el Snapshot 1`] = `
<BrowserRouter>
  <div>
    <NavBar />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/about"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Redirect
        to="/"
      />
    </Switch>
  </div>
</BrowserRouter>
`;
